### PR TITLE
Add guidelines for multiple file upload

### PR DESCRIPTION
**Pull request by mcarrano**
Adds deign guidelines for the [multiple file upload component](https://www.patternfly.org/v4/components/file-upload---multiple).

Closes: #2761

### CODE REVIEW

**Comment by mcarrano** [comment]
@mmenestr this PR is ready for review. I basically just took Megan's text and images and converted them to markdown without reviewing or editing the original as I assumed the content had already been reviewed. But if you see any issues, we can discuss them.

**Comment by mcarrano** [comment]
@mmenestr That's a fair question. But despite the inconsistencies, I kind of like them this way. I'd vote to keep them as they are unless you feel strongly we should change them.

**Comment by mmenestr** [comment]
I kind of like them too, so let's not change them haha the one thing I would change though is the pointers maybe? They are circles instead of the usual pointers we have in our other guidelines

**Comment by mcarrano** [comment]
OK. That's a pretty easy update.

**Comment by mcarrano** [comment]
@mmenestr I updated the image callouts. Hopefully this is all set now. Please take another look.

File: packages/v4/src/content/design-guidelines/components/file-upload-multiple/file-upload-multiple.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+**When to use:**
+* Uploading 1 or more files.
+
+* Uploading files by dragging and dropping.
+
+* Showing the process of uploading. If large file sizes will be supported in your context, showing progress is important. However, if the file size supported will always upload in a matter of a few seconds this may not be necessary.
+
+**Best practices:**
+
+<img src="./img/best-practices.png" alt="best practices for file upload" width="1145"/>
```
Comment (mmenestr) [inline packages/v4/src/content/design-guidelines/components/file-upload-multiple/file-upload-multiple.md:42]: Update image to be 2x PNG instead of 1x PNG!

Comment (mcarrano) [reply packages/v4/src/content/design-guidelines/components/file-upload-multiple/file-upload-multiple.md:42]: I made this update. Please retest once the build completes.

**Review by mmenestr — APPROVED** [review]
Looks great @mcarrano thank you!!

**Pull request merged**